webapp/channels/src/components/widgets/modals/components/checkbox-item-creator.tsx

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import type {SectionItemProps} from './section_item_creator';
+import SectionItemCreator from './section_item_creator';
+
+export type FieldsetCheckbox = {
+    dataTestId?: string;
+    title: {
+        id: string;
+        defaultMessage: string;
+    };
+    name: string;
+}
+
+type Props = SectionItemProps & {
+    inputFieldData: FieldsetCheckbox;
+    inputFieldValue: boolean;
+    handleChange: (e: boolean) => void;
+}
+function CheckboxItemCreator({
```

Comment:
Nitpicking about naming for a moment, could we perhaps name these components `CheckboxSettingItem`, `RadioSettingItem`, etc instead of `CheckboxItemCreator` and such? I think that'd make it a bit clearer on what they're used for since the word "Creator" is confusing me slightly in this context

Reply:
Also, would you mind renaming these files to all use snake case in the file name? A few of them have hyphens instead of underscores (which I think is kebab case)